app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt

```diff
@@ -0,0 +1,73 @@
+package org.p2p.wallet.rpc.interactor
+
+import org.p2p.solanaj.core.Account
+import org.p2p.solanaj.core.Transaction
+import org.p2p.solanaj.programs.TokenProgram
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.rpc.repository.RpcRepository
+import org.p2p.wallet.utils.toPublicKey
+
+class TokenInteractor(
+    private val rpcRepository: RpcRepository,
+    private val addressInteractor: TransactionAddressInteractor,
+    private val tokenKeyProvider: TokenKeyProvider
+) {
+
+    suspend fun close(addressToClose: String): String {
```

Comment:
from the current naming I can't understand what we are closing really

Reply:
For example I can inject your interactor inside my class and use it this way:
```
val interactor : TokenInteractor by inject()

interactor.open(....)
interactor.close(...)
```
it looks like we are closing and opening the interactor, but not the token itself